crates/burn-ndarray/src/ops/base.rs

```diff
@@ -567,12 +567,14 @@ where
             tensor
                 .array
                 .mapv(|x| {
-                    if x > E::zero() {
-                        E::one()
-                    } else if x < E::zero() {
-                        -E::one()
+                    let zero = 0.elem();
+                    let one = 1.elem::<E>();
```

Comment:
Do you think it would be better to declare `zero` and `one` outside the `mapv` ?